src/App.tsx

```diff
@@ -1,19 +1,40 @@
 import React, { useState } from 'react';
 import styled from 'styled-components';
+import { ripOutPaths, generateTree } from './tree';
+import { GithubAPIResponseBody } from './tree/types';
 
 const App: React.FC = () => <Input />;
 
 const Input: React.FC = () => {
-  const [url, setUrl] = useState(
-    'Enter a Github Repo',
-  );
+  const [url, setUrl] = useState('');
+  const [markdownStrings, setMarkdownStrings] = useState(['']);
+
+  const handleChange = event => {
+    event.preventDefault();
+    setUrl(event.target.value);
+  }
 
-  const handleChange = (event) => setUrl(event.target.value);
+  const handleKeyPressed = async event => {
+    if (event.key === 'Enter') {
+      // Expecting a URL like 'github.com/${owner}/${repo}'
+      let pathArray = url.split('/');
+      let owner = pathArray[1];
+      let repo = pathArray[2];
+      await makeRequest(owner, repo);
+    }
+  }
+
+  const makeRequest = async (owner: String, repo: String) => {
+    let response = await fetch(`https://api.github.com/repos/${owner}/${repo}/commits/master`).then(res => res.json());
+    let treeSHA = response["commit"]["tree"]["sha"];
+    let treeRes = await fetch(`https://api.github.com/repos/${owner}/${repo}/git/trees/${treeSHA}?recursive=true`).then(res => res.json());
```

Comment:
We've got `async/await` function calls as well as `.then()` promises. I'm afraid that this may be confusing upon first glance. Is this worth changing to be consistent? I'm probably just being nit-picky.

Reply:
Also, if any of these asynchronous calls fail, or the underlying promises don't resolve, we aren't handling those situations with `try/catch` blocks or `.catch()` calls. We should look into handling possible errors so we can display something on the screen to the user about what happened.

Reply:
@nchaloult are you suggesting this as a better pattern?

```js
let response = await fetch(`https://api.github.com/repos/${owner}/${repo}/commits/master`);
response = response.json();
```

Reply:
@ianjennings I am. I should've been more clear and given and example like you did. What do you think the better pattern is here?

Reply:
Also @ianjennings use a Try Catch to display an error if there is an issue like bad URL etc

Reply:
I've decided to close this pull request. I'd prefer working off of the master branch after merging #12 . It'd save the headache of solving merge conflicts. I'll add the API call function with a `.catch()`

Reply:
👌 If you think that's best, then that sounds good to me